### ISSUE TITLE
Ways to speed up HighExistence

### ISSUE THREAD

**Issue by JonBrooksX** [jetlej/HE-6.0-Bugs#193]
If you type our domain name here: 

https://developers.google.com/speed/pagespeed/insights/

our page isn't on the whole very fast. Should be over 85/100 ideally. 

based on google's recommended actions, are there any ways we can improve site speed?

**Comment by MSchirp** [comment]
Do you have any suggestions for us @dpatil-gmcs ? Can you perhaps clarify specific steps that can be undertaken @JonBrooksX ?

**Comment by dpatil-gmcs** [comment]
@MSchirp : 
We can probably use WP Plugins to achieve Image Optimization and Site Caching
For Image Optimization we can use: https://wordpress.org/plugins/wp-smushit/
For Caching : https://wordpress.org/plugins/wp-super-cache/
Thanks..!!!

**Comment by MSchirp** [comment]
What are your thoughts on these @jetlej ?

**Comment by jetlej** [comment]
@dpatil-gmcs 

We're already setup with W3TC. We've tried many caching plugins and that is what works best. Let's not touch this setup since it's been consistent for a long time now. 

You can re-enable WP Smushit although that plugin is known to hog memory and sometimes cause crashes. I have it disabled on Futurism for this reason. 

The real thing to do here is experiment with combining and minifying JS and CSS via W3TC and making sure nothing breaks. Most of that bad score is from a lack of combining/minification.

**Comment by dpatil-gmcs** [comment]
@jetlej @MSchirp : 

Seems like the W3TC plugin although activated on Live Site, is not working..Reason being insufficient priviliges as shown in the attached image. 

Please provide us the FTP credentials of LIVE Site on Slack, that way we can modify the .htaccess file for live site.

**Comment by jetlej** [comment]
@dpatil-gmcs - That issue is caused by a virus that keeps infecting our server. It works by somehow altering /wp-includes/nav-menu.php to include some base64 code that locks down the .htaccess. I have searched high and low for the source of this virus, but can't find it. 

I'm restoring that nav-menu.php temporarily so the .htaccess file can be changed by W3TC, but it will surely come back again. 

Also, didn't I already give you FTP access?

**Comment by dpatil-gmcs** [comment]
@jetlej : Sure, thanks for the info. You have provided us FTP credentials for Staging / Test Site only and not for the LIVE Site.

**Comment by dpatil-gmcs** [comment]
@MSchirp :
Activated the WP Smush plugin on live site. It is used for Image Optimization. But this might have deactivated by @jetlej during crash recovery. Thanks !

**Comment by MSchirp** [comment]
@dpatil-gmcs It seems the plug in is activated again. Can you see whether speed site has been improved? Does it make sense to upgrade to smush pro? @JonBrooksX what do you think?

**Comment by JonBrooksX** [comment]
The speed score is still in the low 60s on both mobile and desktop. This isn't good enough. Needs to be 80+. 

For mobile, here are some things you can modify @dpatil-gmcs:

![screenshot 2016-01-27 17 42 15](https://cloud.githubusercontent.com/assets/13107838/12622497/92cb8930-c51d-11e5-9253-27fd42886866.png)

For more info, visit: https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Fhighexistence.com&tab=mobile

For desktop, the images are the biggest thing that seems to be slowing down the site. It's $19 for wp-smush pro for one month. I say we get it for one month, smush everything we can within that timeframe @MSchirp. The only alternative is manually going through each image and altering the the size of them. That would take a very long time.

**Comment by MSchirp** [comment]
Just to confirm, @JonBrooksX , if you cancel pro, the images stay smushed?

**Comment by JonBrooksX** [comment]
@MSchirp Yes. Images stay smushed even AFTER you cancel.

**Comment by MSchirp** [comment]
also check out these insights @dpatil-gmcs : https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Fhighexistence.com%2F25-spectacular-movies-you-probably-havent-seen%2F

**Comment by dpatil-gmcs** [comment]
@MSchirp  We have added the setting to the W3 Total Cache plugin to compress the Html, CSS and JS files, but it did not work for us. To optimize the images please use the following tool, https://tinypng.com/
The above tool shrinks the images maintaining the aspect ratio of the image.

**Comment by MSchirp** [comment]
@dpatil-gmcs We have already compressed all images. What needs to be done are the following instructions, it has links on the page on how to fix:

![screen shot 2016-03-04 at 15 45 55](https://cloud.githubusercontent.com/assets/13124056/13530211/82b765ee-e220-11e5-8420-39dc7c9da42a.png)
![screen shot 2016-03-04 at 15 46 41](https://cloud.githubusercontent.com/assets/13124056/13530212/82da9212-e220-11e5-8738-9f353e676b0a.png)

**Comment by dpatil-gmcs** [comment]
@MSchirp  There are still some images which needs to be optimised, Please see the attached screenshot.
For Browser caching and CSS, JS and HTML minification, There are setttings  in W3TotalCache, I have enabled the settings but it seems like these are not working as expected.
Possible reasons: The server is not configured for memcached. Can you please check with you server guy, if the server is configured for Memcached(like Page Cache and minify).
![image_optimize](https://cloud.githubusercontent.com/assets/5171153/13636245/f24653e8-e626-11e5-9216-6959358a42fc.png)

**Comment by JonBrooksX** [comment]
@dpatil-gmcs do you know if there are images uploaded into the media channel or if they're developer images such as logo/icons/etc.?

**Comment by dpatil-gmcs** [comment]
@JonBrooksX @MSchirp 
After looking at the list, I can see that most of the images are user avatar images and other developer images, so I think Compressing the avatar images may increase the score.

**Comment by haydenjames** [comment]
@jetlej The issue with .htaccess being locked has been fixed. It was not a virus but some sort of corrupt config. 

Not sure the exact fixed but the changing of .htaccess permissions has stopped now and so W3TC now has access to write to it.

Here what was done:

Here's what I did.
-- Updated wordpress version.
-- Updated ALL plugins
-- Deleted ALL unused plugins.
-- Deleted w3 cache and then manually deleted ALL of it's backup config files and folders.
-- Reinstalled w3 cache and enabled it.
-- The .htaccess file now does not have the bug of blocking permissions. (fixed)

Now that the .htaccess issue is no more, you can now use W3TC with either memcached (RAM) or disk cache (which will be mounted to tmpfs RAM). Please reconfigure W3TC's Minify and CDN settings.

**Comment by dpatil-gmcs** [comment]
@MSchirp @haydenjames   We have made W3TC minify and cache settings,The score is now improved a bit,
W3TC cannot minify the css that are versioned, so it doesn't minify our style.css file which is like style.css?rel=1457675252946. So we are now researching the ways to minify versioned css files and 3rd party js files, the js files which are loaded by plugin.
Also, Optimization of images is very important for speeding up the site, So can you please optimize the images on the CDN, the list of images can be found in insights results.

**Comment by MSchirp** [comment]
@JonBrooksX Can you take care of the images on CDN with smush?

Also check out the blog posts speeds. @dpatil-gmcs : https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Fhighexistence.com%2F25-spectacular-movies-you-probably-havent-seen%2F

---

@JonBrooksX Can you also check the previous link? It says some images are still big, I thought smush did our whole database?

**Comment by haydenjames** [comment]
Just use Cloudflare's built-in image optimization feature:
https://blog.cloudflare.com/introducing-polish-automatic-image-optimizati/

**Comment by JonBrooksX** [comment]
@MSchirp Smush only reduces file size, so it makes big files **less big** as opposed to small. It also doesn't do developer files.

**Comment by haydenjames** [comment]
Are you guys aware that W3 cache isn't even active anymore? Someone set it into "preview mode".

**Comment by JonBrooksX** [comment]
No, I wasn't aware of this. Is it working now?

**Comment by dpatil-gmcs** [comment]
@MSchirp @haydenjames @JonBrooksX  I can see W3TC active. and I did not find any preview mode enabled option.

**Comment by dpatil-gmcs** [comment]
@MSchirp Just an FYI, I found a plugin, which says that it will bypass the 'render blocking Javascript and CSS'.
I have installed the plugin on the TEST site. For the changes to reflect at google insights, it takes some time. I will verify the Insights result tomorrow and check if it makes any difference.

**Comment by MSchirp** [comment]
Sounds good!

**Comment by JonBrooksX** [comment]
Yeah, this sounds very promising. I really want HE to be blazing fast.

**Comment by dpatil-gmcs** [comment]
@MSchirp  @JonBrooksX  The score seems 75 now, We have invested some time to fix the Leverage Browser Caching,  and found that the problem still shows in the Google Insights results.

We have tried putting the code in the htaccess, tried few plugins like fastest cache etc. but still the problem found. Researching more about the problem and trying to resolve it.

Please let us know your thoughts on it!

**Comment by MSchirp** [comment]
Cool, keep researching :-) We want to hit 100/100 (if that is possible at all)
